test/e2e/base/run.go

```diff
@@ -0,0 +1,38 @@
+package base
+
+import (
+	"errors"
+	"os/exec"
+	"path/filepath"
+	"time"
+)
+
+// RunETCDAdapter start an etcd adapter instance
+func RunETCDAdapter(config string) (*exec.Cmd, error) {
+	p, _ := filepath.Abs("../etcd-adapter")
+	c := exec.Command(p, "-c", config)
+	err := c.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	// Determine if the ETCD Adapter started successfully by
+	// checking HTTP accessibility
+	// The default will retry 5 times, HTTP timeout (1s) time with
+	// waiting time (1s) not more than 10 seconds
+	var remainRetries = 5
+	for {
+		if remainRetries == 0 {
+			return nil, errors.New("exceeds the number of run check retries")
+		}
+		resp, err := httpClient.R().Get("http://" + DefaultAddress)
```

Comment:
I suggest using httpexpect here.

Reply:
I will try it out, I have no experience with httpexpect.

Reply:
Migrated to httpexpect